src/VolumeRenderImpl.ts

```diff
@@ -24,6 +24,12 @@ export interface VolumeRenderImpl {
   setShowBoundingBox: (_show: boolean) => void;
   setBoundingBoxColor: (_color: [number, number, number]) => void;
   viewpointMoved: () => void;
+  /**
+   * Sets the visible z slice, for volume renderers that support it.
+   * @returns true if the z slice was set correctly, false if the slice is out of range.
+   * Defaults to true if z-slice rendering is not implemented on the volume renderer.
+   */
+  setZSlice: (_slice: number) => boolean;
```

Comment:
There is a trap you can fall into here which is that whenever there is a special case suddenly every other subclass has to deal with it because it had to be added to the polymorphic interface, and the interface keeps growing.    I don't have a better solution for now; just noting this.  The refactor has maybe some potential to reduce the number of interface methods.

Reply:
This gets addressed in the VolumeRenderImpl refactor, yeah!